### PR TITLE
Fixes #785 (Adds ability to chain process plugins)

### LINKED ISSUE #785: Chaining processor --> processor in a workflow is not supported
In `processorJob.Run()`, the only parent job type which is available is `collectJob`: https://github.com/intelsdi-x/snap/blob/717b91f0e74ba1377dec0e1c801f18b613a05266/scheduler/job.go#L290-L343

This is a large block; the most important part is [line 332](https://github.com/intelsdi-x/snap/blob/717b91f0e74ba1377dec0e1c801f18b613a05266/scheduler/job.go#L332). This `default` is for the parent job type `switch`.

cc: @nqn

### PR DESCRIPTION
**Pull request by tjmcs**
Fixes #785

Summary of changes:
- Adds code to support process chaining
- Adds a unit test to the `scheduler/workflow_test.go` file to test the new process chaining code
- Adds code to the `scheduler/workflow_test.go` file to catch the `Control.PluginLoaded` event and uses that code to wait for plugins to load in the new unit test (instead of just sleeping for 100ms in order to 'wait' for the plugin to load)

Testing done:
- ran new unit test to prove that two processes can be chained together in a workflow:
```zsh
➜  snap git:(tb/processor-chaining) go test -v --run TestProcessChainingWorkflow ./scheduler/...
=== RUN   TestProcessChainingWorkflow

  Given a started plugin control
    create a workflow with chained processors ✔✔✔✔✔✔✔✔
      Start scheduler ✔
        Create task ✔✔


11 total assertions

--- PASS: TestProcessChainingWorkflow (3.82s)
PASS
ok  	github.com/intelsdi-x/snap/scheduler	3.833s
PASS
ok  	github.com/intelsdi-x/snap/scheduler/wmap	0.010s
➜  snap git:(tb/processor-chaining)
```

@intelsdi-x/snap-maintainers

### CODE REVIEW

File: scheduler/job.go [hunk 1]
```diff
@@ -329,7 +332,32 @@ func (p *processJob) Run() {
 			}).Fatal("unsupported content type")
 			panic(fmt.Sprintf("unsupported content type. {plugin name: %s version: %v content-type: '%v'}", p.name, p.version, p.contentType))
 		}
+	case *processJob:
+		// TODO: Remove switch statement and rely on processor to catch errors in type
+		// (separation of concerns; remove content-type definition from the framework?)
+		switch p.contentType {
+		case plugin.SnapGOBContentType:
```
Comment (geauxvirtual) [inline scheduler/job.go]: Need a default for this switch statement.

**Comment by geauxvirtual** [comment]
Commit message needs to start with `Fixes #785`

**Comment by tjmcs** [comment]
default added to switch and commit message changed to start with "Fixes #785"...thx for the quick feedback @geauxvirtual

**Comment by jcooklin** [comment]
LGTM

**Pull request merged**